README.md

```diff
@@ -1,167 +1,198 @@
-
-Convert this diagram into a table of the form:
-
-Current state | Next word | Next state
---------------|-----------|-----------
-    S0        |    the    |     S1
-    S1        |   . . .   |   . . .
-
-(hint: my version has 13 entries. Yours _might_ be different)
-
-## A2.4
-
-«replace this with your answer»
-
-
-## Q2.5 (12 points)
-
-Translate this table into a programming language of your choice. Then write a
-function that takes a list of words (ending "EOI") and runs the list through the
-state machine. If the state machine cannot find a transition for a word when in
-a given state, return false. If the state machine runs out of words and the
-state is not "END" then return false. Otherwise return true.
-
-Then write some unit tests that exercise your function, making sure that it
-recognizes valid sentences and rejects invalid ones.
-
-The answer should appear in one or more source files in the same directory as
-this file. If I need to do anything more that type a single command to run your
-code, include a script or makefile that will do the job.
-
-## A2.5
-
-«replace this with your answer»
-
-
-## Q2.6 (3 points)
-
-How many valid sentences are there in this language?
-
-## A2.6
-
-«replace this with your answer»
-
-
-## Q2.7 (1 point for the level, 2 for the sentence)
-
-Which is the simplest Chomsky grammar level for this language? In one sentence,
-explain why.
-
-## A2.7
-
-«replace this with your answer»
+# Grammars Assignment (50 points)
+
+This file contains two questions. The first has 2 parts, the second has 7.
+
+For all but questions 2.3 and 2.5 you'll answer questions by editing this file.
+Your answers will appear in the Ax.y section after each question, replacing the
+placeholders.
+
+For question 2.3, you'll need to add an image file to the repository (.png or
+.pdf) and put its file name as the answer.
+
+For question 2.5, you'll write one or more source code files and add them to the
+repository. You will then show the command used to run the files in the answer.
+Do _not_ upload any binary files for this section.
+
+Please do not edit anything outside the answers sections.
+
+
+# Q1: Mr Fussy Builds a Path
+
+# Q1
+
+Mr Fussy wants to build a path using a row of large square pavers. His path is
+one paver wide and an odd number of pavers long. Pavers come in three colors:
+red, green, blue.
+
+Mr Fussy has a rule: the sequence of colors must be symmetrical across the
+length of the path: the last paver must be the same color as the first, the
+second to last the same color as the second, and so on.
+
+This is a valid path:  rgbgr
+So is this: rrr
+
+This is not: rgb
+This is not: rggr    (it must be an odd length)
+
+## Q1.1  (5 points)
+
+Write a Chomsky type 2 grammar that describes any valid path containing at
+least one paver. Use `S` as the start state, and `r`, `g`, `b` as terminals that
+represent the tiles.
+
+## A1.1
+
+S -> r| g | b
+S -> r S r
+S-> g S g
+S -> b S b
+
+r -> "r"
+g -> "g"
+b -> "b" 

```

Comment:
No deduction, but these aren't technically correct, as r, g, and b are already terminals in the grammar (as they are lower case).